redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/Node.kt

```diff
@@ -18,7 +18,8 @@ package app.cash.redwood
 /**
  * A node with properties that can be measured and laid out inside a flexbox.
  */
-public class Node(
+public class Node<T>(
+  public val widget: T,
```

Comment:
My hypothesis is that the responsibilities are better separated like so:

 - Widget: Opaque object from the native widget system.
 - Node: Flexbox-specific configuration of each widget.
 - Adapter: Answers measurement questions about a widget.

The work in front of us is to decide what should be state on Node (such as flexGrow) and what should be queried  (already `minWidth()`, perhaps `visible()` and others?).